### DISCUSSION TITLE
Question (issue) related to co-products

### DISCUSSION THREAD

**Discussion by GOllagnon** [frePPLe/frepple#591] [Q&A] [answered]
Dear Forum,

I send this message today, because I am extensively disocvering frePPLe since a few weeks: for that I am building a "pilot/prototype" which is supposed to model operations in the chemical industry.

The model is rather complex, with many steps, all steps and operations are planning fine ... except one, which is one operation with one product CP in input, 4 products PF1 - PF4 in output (one main product PF1, 3 co-products).

On that SKU, the results of the calculations by frePPLE are not easy to understand (for me ...), because apparently, frePPLe replenishes not all the required quantity, wihch means that starting a moment, there is a backlog which is increasing and is never fulfilled; a huge requirements appears, and it is not fully replensihed, knownig that the capacity is available on the resource, and also in upstream resources and suppliers

I think that everything is setup correctly - as far as I imagine, based on my knowledge right now ... - meanwhile I have one doubt about one pararmeter : item in the table operation.
In the documentation, it is stated that this field can be left blank, as it is systematically filled with the item produced by the operation in simple case, and that behaviour happens ... except "Only in exceptional modeling situations should you worry about setting this field yourself. Eg when an operation produces multiple items".
**So I setup it that way, as PF1**
(Also, I made other tries with PF2 - PF4, all failed).

It is eaxctly my case now.

The resource is bucket_day, but also failed if setup to infinite

I tried different setup ... not concluding.

=> I suppose that I have to setup the "main" product being produced; in my case, there is one "major" product, and 4 co-products ?
=> Is there any particular attention points to focus for such cases ?

Please, let me know...

![Scrennshot_Shortage](https://github.com/frePPLe/frepple/assets/136841819/7a193d3e-9a16-440c-8f67-0664b2174a33)

Thanks & BR,

Guy.

**Comment by jdetaeye** [comment, marked as answer]
> In the documentation, it is stated that this field can be left blank, as it is systematically filled with the item produced by the operation in simple case, and that behaviour happens ... except "Only in exceptional modeling situations should you worry about setting this field yourself. Eg when an operation produces multiple items".

That's a bit outdated. These days the best practice is the other way around: we always set the item on the operation, and in most models (not yours!) we don't create the operation-material record for the produced item.

Some hints:
- Set the the operation.item field to X on the operation which you want to create when there is a requirement for item X.   I.e if I need 100 of item X, which operation do I want to be planned to replenish the required 100?    
- If some of the co-product are really by-products, almost waste, you should consider lowering the priority of those demands. You want to plan the main product first, and then consume up the co-product with the demands for the almost-waste side product.   The demand of the almost-waste product probably shouldn't even be allowed to trigger extra production of the main product.

**Reply by GOllagnon** [reply]
Thanks for the clarification and the best practices !
